### PR TITLE
Fix training gradient underflow in quantization tests

### PR DESCRIPTION
**Pull request by jiqing-feng**
## What does this PR do?

Changes autocast dtype from `float16` to `bfloat16` in `_test_quantization_training`. Float16's limited dynamic range (max ~65504, min subnormal ~5.96e-8) causes gradients to underflow to zero when passing through quantized tensor subclass operations; bfloat16 shares float32's exponent range and avoids this.

### CODE REVIEW

**Comment by jiqing-feng** [comment]
Hi @sayakpaul . Would you please review the PR? Thanks!

File: tests/models/testing_utils/quantization.py [hunk 1]
```diff
@@ -407,7 +407,8 @@ def _test_quantization_training(self, config_kwargs):
         # Step 3: run forward and backward pass
         inputs = self.get_dummy_inputs()
 
-        with torch.amp.autocast(torch_device, dtype=torch.float16):
+        # Use bfloat16 instead of float16 to avoid gradient underflow with quantized layers
```
Comment (sayakpaul) [inline tests/models/testing_utils/quantization.py]: Is this quantization backend agnostic?

Comment (jiqing-feng) [reply tests/models/testing_utils/quantization.py]: Yes, it's backend agnostic in both senses:

1. **Shared code path**: `_test_quantization_training` is in the shared `QuantizationTesterMixin` base class, called by both BNB and TorchAO. BNB training works fine with `bfloat16` too.

2. **Not a quantization-specific issue**: The gradient underflow is a general `float16` limitation — its small exponent range (max ~65504) can't represent the tiny gradients produced during backprop through certain ops. `bfloat16` shares `float32`'s exponent range, avoiding this regardless of the quantization method used.

Comment (sayakpaul) [reply tests/models/testing_utils/quantization.py]: It doesn't happen on all hardware. So, let's guard it for XPU.

Comment (jiqing-feng) [reply tests/models/testing_utils/quantization.py]: fixed.

Comment (jiqing-feng) [reply tests/models/testing_utils/quantization.py]: I need to check it on A100 to see if it's a xpu specific issue.

**Comment by jiqing-feng** [comment]
Let's hold the PR, I will check it after https://github.com/huggingface/diffusers/pull/13540 merged.

**Comment by jiqing-feng** [comment]
The issue also exists on Nvidia A100.
`pytest tests/models/transformers/test_models_transformer_wan.py::TestWanTransformer3DTorchAo::test_torchao_training`

**Comment by jiqing-feng** [comment]
> The issue also exists on Nvidia A100. `pytest tests/models/transformers/test_models_transformer_wan.py::TestWanTransformer3DTorchAo::test_torchao_training`

Hi @sayakpaul . Would you please check it? Thanks!

**Review by sayakpaul — APPROVED** [review]
